app/models/message_summarizer.rb

```diff
@@ -0,0 +1,184 @@
+class MessageSummarizer
+  class MessageSummary
```

Comment:
I feel like this and the subclasses have gotten long enough where they should probably be moved to separate files. It seems weird that the actual class doesn't start until line 89.